src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -118,19 +116,23 @@ public boolean Disconnect(String peerId){
     public boolean  DisconnectIncomingConnections(){
 
         boolean ret = false;
+
         if (mServerSocketList != null) {
-            for(int i = 0; i < mServerSocketList.size(); i++) {
-                BtToServerSocket tmp = mServerSocketList.get(i);
-                if (tmp != null) {
-                    print_debug("Disconnect:::Stop : mBtToServerSocket :" + tmp.getName());
-                    tmp.Stop();
-                    tmp = null;
-                    mServerSocketList.set(i, tmp);
-                    ret = true;
+            synchronized(mServerSocketList) {
+
+                for (int i = 0; i < mServerSocketList.size(); i++) {
+                    BtToServerSocket tmp = mServerSocketList.get(i);
+                    if (tmp != null) {
+                        print_debug("Disconnect:::Stop : mBtToServerSocket :" + tmp.getName());
+                        tmp.Stop();
+                        tmp = null;
+                        mServerSocketList.set(i, tmp);
+                        ret = true;
+                    }
                 }
-            }
 
-            mServerSocketList.clear();
+                mServerSocketList.clear();
+            }
```

Comment:
@DrJukka Sorry to stick this comment here but I can't add comments to GetBluetoothAddress, GetDeviceName or SetDeviceName. In both cases shouldn't the return value be null if we can't get the default adapter? Also rather than using:

```java
if (bluetooth != null) {
   ret = bluetooth.getName();
}

return ret;
```

Wouldn't it be cleaner to just use:
```java
return bluetooth == null ? null : bluetooth.getName();
```